telemetry/telegraf/convert_test.go

```diff
@@ -0,0 +1,72 @@
+package telegraf
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/stretchr/testify/require"
+)
+
+func loadTestData(tb testing.TB, file string) []byte {
+	tb.Helper()
+	content, err := ioutil.ReadFile(filepath.Join("testdata", file+".txt"))
+	require.NoError(tb, err, "expected to be able to read file")
+	require.True(tb, len(content) > 0)
+	return content
+}
+
+func TestConverter_Convert(t *testing.T) {
+	testCases := []struct {
+		Name       string
+		NumMetrics int
+		NumFrames  int
+	}{
+		{Name: "single_metric", NumMetrics: 1, NumFrames: 1},
+		{Name: "same_metrics_same_labels_different_time", NumMetrics: 2, NumFrames: 2},
+		{Name: "same_metrics_different_labels_different_time", NumMetrics: 2, NumFrames: 2},
+		{Name: "same_metrics_different_labels_same_time", NumMetrics: 13, NumFrames: 1},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.Name, func(t *testing.T) {
+			t.Parallel()
+			testData := loadTestData(t, "single_metric")
+			converter := NewConverter()
+			frameWrappers, err := converter.Convert(testData)
+			require.NoError(t, err)
+			require.Len(t, frameWrappers, 1)
+			for _, mf := range frameWrappers {
+				_, err := data.FrameToJSON(mf.Frame(), true, true)
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestConverter_Convert_NumFrameFields(t *testing.T) {
+	testData := loadTestData(t, "same_metrics_different_labels_same_time")
+	converter := NewConverter()
+	frameWrappers, err := converter.Convert(testData)
+	require.NoError(t, err)
+	require.Len(t, frameWrappers, 1)
+
```

Comment:
consider adding a "golden" test -- I find these super userful for understanding input vs output -- and would love your help/advice making the core interface better :)

See:
https://github.com/grafana/grafana/blob/master/pkg/tsdb/influxdb/flux/executor_test.go#L70
Or:
https://github.com/grafana/oas-datasource/blob/fa755294643e56371afe326850aaf76eb0bb6f88/pkg/server/test/server_test.go#L69

Reply:
Will do

Reply:
Done